src/ServiceControl.Audit.Persistence.InMemory/InMemoryPersistence.cs

```diff
@@ -17,6 +20,20 @@ public void Configure(IServiceCollection services)
             services.AddSingleton<IPersistenceLifecycle, InMemoryPersistenceLifecycle>();
         }
 
-        public IPersistenceInstaller CreateInstaller() => new InMemoryPersistenceInstaller();
+        public void ConfigureInstaller(IServiceCollection services)
+        {
+            services.AddHostedService<InMemoryPersistenceInstaller>();
+        }
+    }
+
+    class InMemoryPersistenceInstaller(IHostApplicationLifetime applicationLifetime) : IHostedService
+    {
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            applicationLifetime.StopApplication();
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken) => Task.CompletedTask;
```

Comment:
It feels off to add an empty installer for the sake of calling `applicationLifetime.StopApplication()` and nothing else

Reply:
Yeah this is not needed of we switch in the setup bootstrapper to have a using around the host and then start/stop it. I went this approach for now so that we can see the implications of what we discussed today